datafusion/expr/src/type_coercion/functions.rs

```diff
@@ -311,6 +329,8 @@ fn coerced_from<'a>(
         Utf8 | LargeUtf8 => Some(type_into.clone()),
         Null if can_cast_types(type_from, type_into) => Some(type_into.clone()),
 
+        List(_) if matches!(type_from, FixedSizeList(_, _)) => Some(type_into.clone()),
```

Comment:
Will it be easier if we run Fixedsizelist to List like `coerce_arguments_for_fun` before this type coercion?

Reply:
I am unable to decide between 'coerce_arguments_for_fun' and 'coerce_arguments_for_signature'. Are there specific benefits for each?

Reply:
One is coerced based on function, another is based on signature. If we target to convert from fixedsizelist to list, it seems `coerce_arguments_for_fun` is more suitable.

Reply:
I still think we can have fixed size list to list conversion in one place.
Currently, it converts both in here and `coerce_arguments_for_fun`, right? Is it possible to have it once?

Reply:
move coerce_arguments_for_fun before signature? Convert FixedSizeList to List before signature coercion. Then we don't need to deal with FixedSizeList

Reply:
Yes, we do. 

In `ExprSchemale`(https://github.com/apache/arrow-datafusion/blob/a7cdf605a1e23608e51889988c239613c80fb671/datafusion/expr/src/expr_schema.rs#L91-L105)
It forces the function return type consistent, but I am not sure of its purpose. Maybe @2010YOUY01 could explain it more.

Reply:
I think we dont need this anymore, input types should be handled later on via `coerce_arguments_for_signature`. They are doing the same things

Reply:
Maybe you can move the error handling here
https://github.com/apache/arrow-datafusion/blob/f4fc2639f1d9d1f4dbc73d39990a83f6bf7a725f/datafusion/optimizer/src/analyzer/type_coercion.rs#L560